torch/_dynamo/exc.py

```diff
@@ -227,19 +237,21 @@ class ObservedAttributeError(ObservedException):
 
 observed_exception_map = {
     StopIteration: ObservedUserStopIteration,
+    LookupError: ObservedLookupError,
+    IndexError: ObservedIndexError,
     KeyError: ObservedKeyError,
     AttributeError: ObservedAttributeError,
 }
 
 
-def raise_observed_exception(e, tx):
+def raise_observed_exception(exc_type, tx, *, args=None, kwargs=None):
```

Comment:
```suggestion
def raise_observed_exception(exc_type, tx, *, args=None, kwargs=None) -> typing_extensions.Never:
```
Tells linter that this will always raise an exception